src/data.js

```diff
@@ -310,23 +419,39 @@ Artifact.prototype.json = function() {
  * the Entity will not be deleted if there is an error deleting the underlying
  * artifact.
  */
-Artifact.prototype.remove = function(ignoreError) {
+Artifact.prototype.remove = async function(ignoreError) {
   // Promise that deleted underlying artifact, and keep reference to context
   var deleted = Promise.resolve();
 
+  // Shorthand because we refer to this property so many times.  Saves
+  // prototypechain lookups and linespace
+  let dtl = this.details;
```

Comment:
Destructuring would make it more complicated, imo, and would require destructuring in three different places (once for blob, once for s3 and once for azure).  I'll rename to details.